include/CLI/App.hpp

```diff
@@ -523,133 +536,157 @@ class App {
     }
 
     /// Set a help flag, replace the existing one if present
-    Option *set_help_flag(std::string flag_name = "", std::string help_description = "") {
+    Option *set_help_flag(std::string flag_name = "", const std::string &help_description = "") {
+        // take flag_description by const reference otherwise add_flag tries to assign to help_description
         if(help_ptr_ != nullptr) {
             remove_option(help_ptr_);
             help_ptr_ = nullptr;
         }
 
         // Empty name will simply remove the help flag
         if(!flag_name.empty()) {
-            help_ptr_ = add_flag(flag_name, std::move(help_description));
+            help_ptr_ = add_flag(flag_name, help_description);
             help_ptr_->configurable(false);
         }
 
         return help_ptr_;
     }
 
     /// Set a help all flag, replaced the existing one if present
-    Option *set_help_all_flag(std::string help_name = "", std::string help_description = "") {
+    Option *set_help_all_flag(std::string help_name = "", const std::string &help_description = "") {
+        // take flag_description by const reference otherwise add_flag tries to assign to flag_description
         if(help_all_ptr_ != nullptr) {
             remove_option(help_all_ptr_);
             help_all_ptr_ = nullptr;
         }
 
         // Empty name will simply remove the help all flag
         if(!help_name.empty()) {
-            help_all_ptr_ = add_flag(help_name, std::move(help_description));
+            help_all_ptr_ = add_flag(help_name, help_description);
             help_all_ptr_->configurable(false);
         }
 
         return help_all_ptr_;
     }
 
-    /// Add option for flag
-    Option *add_flag(std::string flag_name, std::string flag_description = "") {
-        CLI::callback_t fun = [](CLI::results_t) { return true; };
-        Option *opt = add_option(flag_name, fun, flag_description, false);
-        if(opt->get_positional())
-            throw IncorrectConstruction::PositionalFlag(flag_name);
+  private:
+    /// internal function for adding a flag
+    Option *_add_flag_internal(std::string &flag_name, CLI::callback_t &fun, std::string &flag_description) {
```

Comment:
Why non-const `std::string &` for `flag_description`?

Reply:
it uses non-const reference to be able to move the name, description, and function and not copy them first.  We can revisit that but I wanted to avoid an extra copy if possible.

Reply:
If this is something that a user types one by one, a computer is fast enough to make a copy. If it happens inside a loop, then avoiding copies is (more) important. Also, for things like names, I expect those are mostly short strings, and the short string optimization means moves are just as expensive as copies. Keeping the code simple and maintainable is also important.